rero_ils/modules/holdings/listener.py

```diff
@@ -50,3 +59,75 @@ def enrich_holding_data(sender, json=None, record=None, index=None,
                 break
         json['public_items_count'] = \
             json['items_count'] - number_of_masked_items
+
+
+def update_items_locations_and_types(sender, record=None, **kwargs):
+    """This method checks if the items of the parent record needs an update.
+
+    This method checks the location and item_type of each item attached to the
+    holding record and update the item record accordingly.
+    This method should be connect with 'after_record_update'.
+    :param record: the holding record.
+    """
+    if record.get('$schema') != get_schema_for_resource(Holding):
+        return
+    if record.get('holdings_type') == HoldingTypes.SERIAL:
+        # identify all items records attached to this serials holdings record
+        # with different location and item_type.
+        hold_circ_pid = record.circulation_category_pid
+        hold_loc_pid = record.location_pid
+        search = ItemsSearch().filter('term', holding__pid=record.pid)
+        item_hits = search.\
+            filter('bool', should=[
+                        Q(
+                            'bool', must_not=[
+                                Q('match', item_type__pid=hold_circ_pid)
+                            ]
+                        ),
+                        Q(
+                            'bool', must_not=[
+                                Q('match', location__pid=hold_loc_pid)
+                            ]
+                        )
+            ]
+            ).source(['pid'])
+        items = [hit for hit in item_hits.scan()]
+        item_ids = []
```

Comment:
It should be possible to extract the ids directly here with:
`items_ids = [hit.metadata.id for hit in item_hits.scan()]`